### ISSUE TITLE
[UI/UX] PartDesign: Move Object after other Object

### ISSUE THREAD

**Issue by leoheck** [FreeCAD/FreeCAD#16340]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

To find this menu: Right click on any object inside a PartDesign Body

First of all, this feature could be renamed to `Move Object Internally` since there are other operation named `Move Object to other Body`

When using the `Move Object after other Object`, a small popup with a drop-down list of Objects is displayed. 

![image](https://github.com/user-attachments/assets/3a76057c-975b-40a2-bd5a-5ca201285a5c)

The list of objects has just the name (Label) of the objects.
It could have the Object icon too, to make the process of finding this object easier. 
This will improve the usage specially when the order of this list does not reflect the order of the Objects Tree, which is almost always.

Took a picture of the lists since my screenshot tool to capture with a delay is not working right now.

![image](https://github.com/user-attachments/assets/29d796a8-64b2-4e0c-b374-c7ae61908104)

My suggestion is to have something like this with colors, but using just a single level, of course.

![image](https://github.com/user-attachments/assets/ca788718-aaba-41d4-bebe-de682eaf239e)

### Full version info

```shell
OS: Ubuntu 24.04.1 LTS (ubuntu:GNOME/ubuntu-xorg)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38622 (Git) AppImage
Build type: Release
Branch: main
Hash: ed97f3c4591303ee3b54906768f8f4eba5211265
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
```

### Subproject(s) affected?

PartDesign

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
Maybe this should also be possible by drag and drop? @PaddleStroke

**Comment by leoheck** [comment]
Ah, drag and drop would be interesting, indeed but it has a potential to mess things hard since things inside the Body have tight relationship.

**Comment by PaddleStroke** [comment]
I think both Move "Object after other Object" and "Move Object to other Body" tools should be removed and these operations should be handled by drag and drop.
The drag and drop from and to a body just need to have the same rules as the current tools have. If the rules to accept are the same as current tools then I don't see why we would keep the tool no?

**Comment by leoheck** [comment]
I agree, except that it is a bit hard to drag and drop things to the right place I want specially when this place is the last position of a nested object; the first position after a nested object; or even the last position of the object tree. This issue might may be due to the theme (OpenTheme) I am using, but I I feel it was like this since long time ago. This is also a hard thing to do when working on a huge list of objects and features.

**Comment by arf20** [comment]
Bumping this
